pkg/skaffold/event/v2/logger.go

```diff
@@ -54,3 +57,47 @@ func (ev *eventHandler) handleSkaffoldLogEvent(e *proto.SkaffoldLogEvent) {
 		},
 	})
 }
+
+type logHook struct{}
+
+func NewLogHook() logrus.Hook {
+	return logHook{}
+}
+
+func (h logHook) Levels() []logrus.Level {
+	return []logrus.Level{
+		logrus.PanicLevel,
+		logrus.FatalLevel,
+		logrus.ErrorLevel,
+		logrus.WarnLevel,
+		logrus.InfoLevel,
+		logrus.DebugLevel,
+	}
+}
+
+func (h logHook) Fire(entry *logrus.Entry) error {
+	handler.handleSkaffoldLogEvent(&proto.SkaffoldLogEvent{
+		Level:   levelFromEntry(entry),
+		Message: entry.Message,
+	})
+	return nil
+}
+
+func levelFromEntry(entry *logrus.Entry) enums.LogLevel {
+	switch entry.Level {
+	case logrus.FatalLevel:
+		return enums.LogLevel_FATAL
+	case logrus.ErrorLevel:
+		return enums.LogLevel_ERROR
+	case logrus.WarnLevel:
+		return enums.LogLevel_WARN
+	case logrus.InfoLevel:
+		return enums.LogLevel_INFO
+	case logrus.DebugLevel:
+		return enums.LogLevel_DEBUG
+	case logrus.PanicLevel:
+		return enums.LogLevel_PANIC
+	}
+
+	return enums.LogLevel_DEBUG
```

Comment:
super nit: if all branches of a `switch` or `if` statement return, then we don't need a `return foo` at the function end. So you could move this to the `default` `switch` `case` and do away with `logrus.DebugLevel` case.

```go
func levelFromEntry(entry *logrus.Entry) enums.LogLevel {
	switch entry.Level {
	case logrus.FatalLevel:
		return enums.LogLevel_FATAL
	case logrus.ErrorLevel:
		return enums.LogLevel_ERROR
	case logrus.WarnLevel:
		return enums.LogLevel_WARN
	case logrus.InfoLevel:
		return enums.LogLevel_INFO
	case logrus.PanicLevel:
		return enums.LogLevel_PANIC
	default:
		return enums.LogLevel_DEBUG
	}
}
```